sdk/storage/Azure.Storage.Files.Shares/tests/ShareClientTests.cs

```diff
@@ -734,6 +735,51 @@ public async Task Delete_Error()
                 e => Assert.AreEqual("ShareNotFound", e.ErrorCode));
         }
 
+        [Test]
+        [ServiceVersion(Min = ShareClientOptions.ServiceVersion.V2019_12_12)]
+        public async Task RestoreAsync()
+        {
+            // Arrange
+            ShareServiceClient service = GetServiceClient_SoftDelete();
+            string shareName = GetNewShareName();
+            ShareClient share = InstrumentClient(service.GetShareClient(shareName));
+            await share.CreateAsync();
+            await share.DeleteAsync();
+            IList<ShareItem> shares = await service.GetSharesAsync(states: ShareStates.Deleted).ToListAsync();
+            ShareItem shareItem = shares.Where(s => s.Name == shareName).FirstOrDefault();
+
+            // It takes some time for the Share to be deleted.
+            if (Mode != RecordedTestMode.Playback)
+            {
+                await Task.Delay(30000);
```

Comment:
Would be better to wait on/poll (with exponential retry) some API for delete condition instead of hard-coded wait time. Might give more stability in live mode.

Reply:
Could we just keep calling Delete periodically until we stop getting `ShareBeingDeleted` or whatever the error code is?  If you do keep this as-is, we've got a handy https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs#L354 that encapsulates the mode check.

Reply:
Calling delete multiple tests doesn't work, delete doesn't return an error code if a share is being deleted.  I'll switch to using the Delay method.

Reply:
I think we can change this before merging to master to calling Restore() in a loop and waiting for no error, for stability in the live tests.